src/ts/tests/circular_buffer.spec.ts

```diff
@@ -166,6 +166,69 @@ test('[UNIT] at/1', () => {
 
 
 
+test('[UNIT] fill/5', () => {
+
+  // declare a five item cb
+  const filler = new circular_buffer<any>(5);
+
+  // can accept five items
+  expect(filler.fill(1)).toEqual([1,1,1,1,1]);
+});
+
+
+
+
+
+test('[UNIT] fill/full/5', () => {
+  // declare a five item cb
+  const filler = new circular_buffer<any>(5);
+
+  filler.push(1);
+  filler.push(1);
+  filler.push(1);
+  filler.push(1);
+  filler.push(1); 
+  /*we're full*/
+
+  expect(filler.at(0)).toEqual(1);
+
+  expect(filler.at(4)).toEqual(1);
+
+  filler.fill(2);
+  /*fill() will overwrite*/
+
+  expect(filler.at(0)).toEqual(2);
+
+  expect(filler.at(4)).toEqual(2);
+  /*arcaic, but tests ok*/
+
+});
+
+
+
+
+
+test('[UNIT] fill/partial/3', () => {
+  // declare a three item cb
```

Comment:
nuh-uh

Reply:
This particular flag has helped me identify something wrong.  After the fix, a second fix was made